### DISCUSSION TITLE
How to update an array inside an array?

### DISCUSSION THREAD

**Discussion by ghost** [jenssegers/laravel-mongodb#2482] [Q&A]
I have a PurchaseRequest (like a Cart) and inside I have products...
```

{
    "id": "62862f04375eed096c6abc32",
    "products": [
        {
            "id": "628667dccee1aa7afe760d24",
            "currency": "MXN",
            "unit_cost": 10,
            "product_id": 1,
            "quantity": 1,
            "metadata": []
        }
    ]
}
```

I am not able to update just the position 0 of the array products.

I tried with:

`$purchaseRequest->update(['products' => [$currentProductIndex => $currentProduct]]);`

But this replaces the whole products.

I tried with:

`$purchaseRequest->update(['products.' . $currentProductIndex => $currentProduct]);`

But it doesn't do anything.

**Comment by JorgeSivil** [comment]
I've solved it by debugging but I came across another bug...

```
PurchaseRequest::where(
    [
        ['_id', '=', new ObjectId($purchaseRequestId)],
        ['products.id', '=', $productId]
    ]
)->update(
    [
        '$set' => [
            "products.$" => $currentProduct,
        ]
    ]
);
```

I am using the timestamps for update, created and deleted and it is causing the following error, setting $set inside $set

src/Query/Builder.php

![image](https://user-images.githubusercontent.com/12187148/169732454-d15de33c-1151-4239-8469-dbf9dc870d45.png)

![image](https://user-images.githubusercontent.com/12187148/169732498-ba4a3a97-69dd-4cee-bc40-b66c7f9b65ae.png)


So I had to remove the $set inside my update statement, and leave only `products.$`
